### DISCUSSION TITLE
CCU3 -> HB-RF-ETH

### DISCUSSION THREAD

**Discussion by OP-User-AJ** [OpenCCU/OpenCCU#2104] [🇩🇪 Fragen&Antworten] [answered]
Hallo,

ich würde eine ccu3 gerne Virtualisieren, dieser Prozess ist ist sehr gut beschrieben.
https://github.com/jens-maus/RaspberryMatic/wiki/Installation#grundinstallation-virtuell
Nun brauche ich ja nach der Virtualisierung zugriff auf eine Funk-Modulplatine
https://github.com/jens-maus/RaspberryMatic/wiki/Einleitung#vorraussetzungen
Auch das ist da sehr gut beschieben.

Jetzt stellte sich mir die Frage, kann ich die CCU3 so um/neu installieren das diese als HB-RF-ETH ansprechbar ist?
Wenn nein, kann die HB-RF-ETH mit dem Modul aus der CCU3 Betreiben werden?
Oder gibt es eine bessere Lösung RaspberryMatic virtuell zu betreiben ohne Zugriff auf einen USB Port.

**Comment by jens-maus** [comment, marked as answer]
> Jetzt stellte sich mir die Frage, kann ich die CCU3 so um/neu installieren das diese als HB-RF-ETH ansprechbar ist?

Nein. Das geht nicht.

> Wenn nein, kann die HB-RF-ETH mit dem Modul aus der CCU3 Betreiben werden?

Ja, das `RPI-RF-MOD` Funkmodul das in einer CCU3 drinsteckt kann man für die Nutzung mit einem `HB-RF-ETH` wiederverwenden. Einfach ausbauen und auf das `HB-RF-ETH` stecken und es sollte gehen.

> Oder gibt es eine bessere Lösung RaspberryMatic virtuell zu betreiben ohne Zugriff auf einen USB Port.

Wenn du definitiv keinen USB Port durchleiten kannst/willst dann ist das momentan die einzige verlässliche Art&Weise um ein `RPI-RF-MOD` oder `HM-MOD-RPI-PCB` Funkmodul via LAN an ein RaspberryMatic anzubinden. Trotzdem solltest du noch einmal deine Möglichkeiten ggf. evaluieren. Die Anbindung via eine `HmIP-RFUSB` USB Sticks oder nutzung eines `HB-RF-USB-2` ist technologisch einfacher und unabhängig von lokalen Netzwerkproblemen. Und alle gängigen Virtualisierungslösungen erlauben eigentlich problemlos das durchleiten dieser USB-basierten Lösung an die jeweilige RasperryMatic VM.

**Reply by OP-User-AJ** [reply]
Danke!!!!

**Comment by FrankGiesecke** [comment]
Hallo @jens-maus,
besteht auch die Möglichkeit, zwei `HB-RF-ETH` (mit je einem eigenen `RPI-RF-MOD` Funkmodul) in RaspberryMatic zu registrieren?

Es geht mit um die Möglichkeit eines Failover, niert um die zeitgleiche Nutzung der Funkmodule.
Meine RaspberryMatic Instanz läuft virtualisiert auf einem Proxmox Cluster und wird bei Ausfall des Nodes automatisch auf einen anderen Node migriert.
Nur das Funkmodul macht mir noch sorgen. Sollte dies ausfallen, muss aktuell manuell eingegriffen werden. Es muss vom aktiven Modul auf das Standby-Modul umgesteckt werden.

Eine Failover würde hier das System ausfallsicherer gestalten.
Ist so ein Szenario möglich/umsetzbar?

**Reply by jens-maus** [reply]
> Hallo @jens-maus, besteht auch die Möglichkeit, zwei `HB-RF-ETH` (mit je einem eigenen `RPI-RF-MOD` Funkmodul) in RaspberryMatic zu registrieren?
> 
> Es geht mit um die Möglichkeit eines Failover, niert um die zeitgleiche Nutzung der Funkmodule. [...]
> 
> Eine Failover würde hier das System ausfallsicherer gestalten. Ist so ein Szenario möglich/umsetzbar?

Nein, das ist nicht möglich und nicht umsetzbar. Die Funkmodule funktionieren nach dem Motto "Es kann nur einen geben". Auch wenn man mehrere HB-RF-ETH parallel integrieren könnte, würde das System nicht ohne Ausfallzeiten funktionieren. Das liegt daran, dass ein Funkmodulwechsel (d.h. wenn sich die Seriennummer und damit der Verschlüsselungskey des aktuell genutzten Funkmoduls ändert) ein „Rekeying“ erfordert, das derzeit nur über eine Online-Verbindung zu eQ3-Servern möglich ist. Daher kann es nach einem Funkmodulwechsel Minuten bis Stunden dauern, bis alle angelernten HmIP-Geräte wieder funktionieren. Wenn man nun ein automatisches Failover einrichtet, das das Funkmodul ständig wechselt, würde das System am Ende gar nicht mehr funktionieren. Daher ist es nicht nur unmöglich, sondern auch nicht erstrebenswert.

**Reply by FrankGiesecke** [reply]
Alles klar. Vielen Dank für die ausführliche Erklärung. Das macht einiges klarer.
Dann werde ich es bei einer Cold-Standby Lösung belassen.
